module/zfs/zfs_ioctl.c

```diff
@@ -1577,15 +1572,17 @@ zfs_ioc_pool_create(zfs_cmd_t *zc)
 	 * Set the remaining root properties
 	 */
 	if (!error && (error = zfs_set_prop_nvlist(spa_name,
-	    ZPROP_SRC_LOCAL, rootprops, NULL)) != 0)
+	    ZPROP_SRC_LOCAL, rootprops, NULL)) != 0) {
 		(void) spa_destroy(spa_name);
+		unload_wkey = B_FALSE; /* spa_destroy() unloads wrapping keys */
```

Comment:
Personally, I would move this up just beneath `spa_create()`. The issue here is that after `spa_create()` runs (without error), it "owns" the dsl crypto key so we should set this variable as soon as that is true (to prevent future problems). This code is equivalent though.